### ISSUE TITLE
[2022-11-24] Theme breaks some GTK applications

### ISSUE THREAD

**Issue by megascrapper** [Fausto-Korpsvart/Gruvbox-GTK-Theme#24]
After installing the 2022-11-24 update, when I applied the GTK3 theme in GNOME Tweaks (in the `Legacy Applications` field), some apps that use GTK3/4 in some way would either open slowly or just times out and not open at all (e.g. nautilus). In addition, some of these apps would have huge CPU usage for no apparent reason until I killed them.

![Screenshot from 2022-11-27 00-39-15](https://user-images.githubusercontent.com/34503494/204092111-cb6af3ab-8dd6-44a8-8a3a-5bc13d1b8601.png)

OS: Arch Linux

Tested variant: Gruvbox-Dark-BL

Some package versions:

```
gtk2 2.24.33-2
gtk3 1:3.24.35-2
gtk4 1:4.8.2-1
gnome-shell 1:43.1-2
nautilus 43.0-1
gnome-tweaks 42beta+r14+g2d9d76c-2
```

**Comment by Fausto-Korpsvart** [comment]
> 

That's quite strange, I have not seen any performance error in the tests I've done, I'm on Fedora 37, and I've tested on Linux Mint, Arch Linux, Manjaro, MX Linux, Ubuntu, PopOS and even on Fedora 36 and I have not noticed bad performance in any distro, I will check carefully because I do not think it is due to the themes, since the tests I've done are in clean virtual installations without extensions, as they come by default and I have not noticed problems; but I repeat, I'm going to check carefully to find out what could be the cause, in case it's the themes.

Thanks for reporting and for the support }:-]

**Comment by megascrapper** [comment]
Hi, thanks for your response (and awesome work on the theme)

I tried to reproduce the bug on Kali Linux with the same `gnome-shell` version (but probably slightly different GTK versions) and the issue didn't appear. I'll try some other OSes with a clean state if I have the time.

With that being said, I managed to capture some more things with my Arch install:

- Logging in with the GTK3 theme loaded under Wayland would cause `xdg-desktop-portal.service` and `xdg-desktop-portal-gnome.service` to fail to start (`xdg-desktop-portal-gtk.service` is fine though, not sure why). This causes apps/extensions that rely on it (e.g. [DING](https://extensions.gnome.org/extension/2087/desktop-icons-ng-ding/)) to bootloop.

```
$ systemctl --no-pager -l status --user xdg-desktop-portal.service
× xdg-desktop-portal.service - Portal service
     Loaded: loaded (/usr/lib/systemd/user/xdg-desktop-portal.service; static)
     Active: failed (Result: timeout) since Sun 2022-11-27 12:13:07 AEDT; 1min 4s ago
    Process: 114988 ExecStart=/usr/lib/xdg-desktop-portal (code=killed, signal=TERM)
   Main PID: 114988 (code=killed, signal=TERM)
        CPU: 50ms

Nov 27 12:11:37 fadi-surface systemd[110578]: Starting Portal service...
Nov 27 12:12:27 fadi-surface xdg-desktop-por[114988]: Failed to create settings proxy: Error calling StartServiceByName for org.freedesktop.impl.portal.desktop.gnome: Timeout was reached
Nov 27 12:12:53 fadi-surface xdg-desktop-por[114988]: Failed to create file chooser proxy: Error calling StartServiceByName for org.freedesktop.impl.portal.desktop.gnome: Timeout was reached
Nov 27 12:12:53 fadi-surface xdg-desktop-por[114988]: No skeleton to export
Nov 27 12:13:07 fadi-surface systemd[110578]: xdg-desktop-portal.service: start operation timed out. Terminating.
Nov 27 12:13:07 fadi-surface systemd[110578]: xdg-desktop-portal.service: Failed with result 'timeout'.
Nov 27 12:13:07 fadi-surface systemd[110578]: Failed to start Portal service.
```

```
$ systemctl --no-pager -l status --user xdg-desktop-portal-gnome.service
× xdg-desktop-portal-gnome.service - Portal service (GNOME implementation)
     Loaded: loaded (/usr/lib/systemd/user/xdg-desktop-portal-gnome.service; static)
     Active: failed (Result: timeout) since Sun 2022-11-27 12:22:50 AEDT; 11s ago
    Process: 2415 ExecStart=/usr/lib/xdg-desktop-portal-gnome (code=killed, signal=TERM)
   Main PID: 2415 (code=killed, signal=TERM)
        CPU: 1min 30.464s

Nov 27 12:21:20 fadi-surface systemd[1488]: Starting Portal service (GNOME implementation)...
Nov 27 12:22:50 fadi-surface systemd[1488]: xdg-desktop-portal-gnome.service: start operation timed out. Terminating.
Nov 27 12:22:50 fadi-surface systemd[1488]: xdg-desktop-portal-gnome.service: Failed with result 'timeout'.
Nov 27 12:22:50 fadi-surface systemd[1488]: Failed to start Portal service (GNOME implementation).
Nov 27 12:22:50 fadi-surface systemd[1488]: xdg-desktop-portal-gnome.service: Consumed 1min 30.464s CPU time.
```

- On X, the two services can start, but the slow/hanging start for some apps still occur
- Disabling all extensions doesn't seem to make a difference
- I tried reinstalling `xdg-desktop-portal` and `xdg-desktop-portal-gnome` but it doesn't make a difference either

**Comment by megascrapper** [comment]
Update: looks like I managed to solve it!

I think the problem is that I symlinked the entire `gtk-4.0` folder to `~/.config` instead of symlinking the contents to the existing `~/.config/gtk-4.0`, though I can never be so sure.

I also patched `gtk.css` and `gtk-dark.css` to fix a somewhat related bug in #23. One bug remained though: looks like you swapped the `gtk.css` files between the -LB and non -LB variant as I get the legacy window titles then linking `~/.themes/Gruvbox-Dark-BL/gtk-4.0/*`. I'll make a separate PR if I have the time.

**Comment by Fausto-Korpsvart** [comment]
> 



> 
Sorry for the delay in responding.

I did the test by linking the whole `gtk-4.0` folder to `.config` and I noticed a performance drop, that's why you need to link only the content inside the `gtk-4.0` folder of the theme to the `gtk-4.0` folder in `.config`, that solves the performance issue.

With the Gruvbox-BL Theme and the LB version, I will update soon, as I have made the file change and placed them in their corresponding Theme.

As always, thanks for the support and help with theme improvements.
